docs/motivation.md

```diff
@@ -0,0 +1,550 @@
+# The Verona language: Core type system
+
+Verona is a language with a behaviour-oriented concurrency model and a region abstraction for ownership.
+These are fundamentally new abstractions in language design and are covered separately.
+This document aims to cover the parts of the type system that are similar to (though in some places diverge in interesting ways from) other languages.
+
+Verona is an object oriented reference capability language with structural
+subtyping and bounded polymorphism. This document aims to describe the aspects
+of the type system that are independent from both the concurrency model and
+reference capabilities.
+
+## Structural type system
+
+A type system is all about managing a complexity budget. Make it too complicated
+and programmers will find it difficult to use and be less productive. Make it too
+simple and it will give you less guarantees to lean on as a programmer.
+
+We opt to use a structural type system since we beleive it simplifies some parts
+of reasoning, e.g. we don't need co/contra-variance notation on type parameters.
+This allows us to spend this complexity elsewhere, e.g. introducing reference
+capabilities to enforce isolation and region topology.
+
+We do however treat classes nominally and have opted to not have inheritance
+subtyping. This means that there is never any subtyping between two distinct
+classes.
+
+### Traits
+Verona is a structural type system based on traits, i.e. types can be described
+using combination of methods on an object. E.g. we can describe a type
```

Comment:
This is so minor, but: "e.g." is "for example" in Latin, requires a comma after it, and is bad form to use at the beginning of a sentence. Here it can be "object, e.g., we". "i.e." is "inter alia" and also requires a comma. Using both in the same sentence is an indicator that you probably had an entirely different structure in mind and you might want to rewrite :)

Reply:
Is the comma issue a British vs. American English thing? I agree on not starting a sentence with either though.

Reply:
I think I would put comma's in like Sylvan says, e.g., like in this sentence.
But mostly, I use
> "for example" instead of "e.g."

and

> "that is" instead of "i.e."